chain/chain/src/chain.rs

```diff
@@ -1088,9 +1090,17 @@ impl Chain {
             &vec![challenge.clone()],
             &head.epoch_id,
             &head.last_block_hash,
-            None,
         ) {
-            Ok(_) => {}
+            Ok((_, challenged_blocks)) => {
+                for block_hash in challenged_blocks {
+                    match chain_update.mark_block_as_challenged(&block_hash, None) {
+                        Ok(()) => {}
+                        Err(err) => {
+                            warn!(target: "chain", "Error saving block as challenged: {:?} {:?}", block_hash, err);
```

Comment:
Nit, please use tracing key-values instead of formatting information into the message

```suggestion
                            warn!(target: "chain", ?block_hash, ?err, "Error saving block as challenged");
```